backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/pporder/process/WEBUI_PP_Order_PrintLabel.java

```diff
@@ -56,15 +47,28 @@
 import de.metas.ui.web.pporder.PPOrderLineType;
 import de.metas.ui.web.window.datatypes.DocumentIdsSelection;
 import de.metas.util.Services;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.trx.api.ITrx;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_M_Product;
+import org.compiere.model.I_M_Product_PrintFormat;
+import org.compiere.util.DB;
 import org.springframework.core.io.ByteArrayResource;
 
-public class WEBUI_PP_Order_PrintLabel extends WEBUI_PP_Order_Template implements IProcessPrecondition
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+public class WEBUI_PP_Order_PrintLabel extends WEBUI_PP_Order_Template implements IProcessPrecondition, IProcessDefaultParametersProvider
 {
 	private static final String PARAM_AD_PrintFormat_ID = "AD_PrintFormat_ID";
 
 	@Param(parameterName = PARAM_AD_PrintFormat_ID)
 	private int printFormatId;
 
+	@Param(parameterName = I_M_Product.COLUMNNAME_M_Product_ID)
+	private ProductId productId;
```

Comment:
where is this parameter used?

Reply:
It's used to provide a value for the AD_Val_Rule of the print format. In case a HU is selected, the view row context will not properly resolve the M_Product_ID by itself. So this will ensure an `M_Product_ID` is in the evaluation context.